java-manta-client/src/main/java/com/joyent/manta/client/multipart/EncryptionStateSnapshot.java

```diff
@@ -83,8 +106,8 @@ public boolean equals(final Object o) {
         }
 
         final EncryptionStateSnapshot that = (EncryptionStateSnapshot) o;
-        return lastPartNumber == that.lastPartNumber
-                && cipher == that.cipher
-                && hmac == that.hmac;
+        return Objects.equals(uploadId, that.uploadId)
```

Comment:
Sorry if I breezed by on the last patch, why does this class needs equals & hashcode methods.